README.md

```diff
@@ -21,40 +21,52 @@ Currently the docker-compose performance isn't very good. Thus, for local dev, i
+1. First, we need to install poetry:
 
-```
-    export PYTHON_VERSION=3.8.13
-    pyenv install $PYTHON_VERSION
-    pyenv global $PYTHON_VERSION
-    pyenv virtualenv $PYTHON_VERSION workflows
-    pyenv activate workflows
-```
+    ```
+    curl -sSL https://install.python-poetry.org | python3 -
 
-3. Set airflow home directory:
+    ```
 
-```
-export AIRFLOW_HOME=/path/to/cloned/repo
-```
-
-4. Install all required dependencies for a project. We need 3 requirements files, when we're running the project locally. First file (requirements.py) has all additional dependencies required to run the tasks correctly (connecting to ftputil, boto3 and etc.), the second one installs requirememnts for testing, such as pytest, the third- (requirements_airflow.py) installs Airflow itself and constraints needed for it. The the third file is not needed when we're running the procject on Docker, because we're using Apache Airflow Docker image:
-   `pip install -r requirements.txt -r requirements-test.txt -r requirements-airflow.txt`
-5. Run Airflow. Airflow comes with the `standalone` command. This should not be used as it seems to use the SequentialExecutor by default. To have our local config being use, we must run all services together :
+3. Set airflow home directory:
 
-```
-    airflow webserver
-    airflow triggerer
-    airflow scheduler
-```
+    ```
+    export AIRFLOW_HOME=/path/to/cloned/repo
+    ```
+4. Activate poetry virtual environment:
+    ```
+    poetry shell
+    ```
+    If you are using incorrect python version, you will get an error:
+    ```
+    current Python version (X.Y.Z) is not allowed by the project (^X.Y).
+    Please change python executable via the "env use" command.
+    ```
+    You can easily fix it with the following command:
+    ```
+    poetry env use X.Y
+    ```
+    The correct python version can be found in pyptoject.toml file
+5. Install dependencies listed in pyproject.toml file by running the command:
+    ```
+    poetry install
+    ```
+6. Run Airflow. Airflow comes with the `standalone` command. This should not be used as it seems to use the SequentialExecutor by default. To have our local config being use, we must run all services together :
+
+    ```
+        poetry run airflow webserver
+        poetry run airflow triggerer
+        poetry run airflow scheduler
+    ```
 
 ### Script
 
 A Makefile has been created to ease this process. The available targets are the following :
 
-- `make init` : Inits the pyenv version and virtual env.
-- `make start` : Starts all processes needed.
-- `make stop` : Stops all processes needed.
-- `make create_user` : create a new user with all the parameters
+- `poetry run make init` : Inits the pyenv version and virtual env.
+- `poetry run make start` : Starts all processes needed.
+- `poetry run make stop` : Stops all processes needed.
+- `poetry run make create_user` : create a new user with all the parameters
```

Comment:
`poetry` and `make` are two independent tools, I guess what we want to achieve here is running with poetry the commands **inside** make, hence we should update `Makefile` instead. :)

Reply:
Dockerfile - yes. The reason is I want to see if airflow runs by using poetry. That's why I am taking the Python image, installing poetry, and then running tests by using this container. 

https://github.com/cern-sis/workflows/blob/9a0124e82d48c505f910783547daa7ec6dc8532c/.github/workflows/test-and-build.yml#L72-L76 (the file is from Poetry PR)